pkg/koordlet/statesinformer/config.go

```diff
@@ -16,27 +16,31 @@ limitations under the License.
 
 package statesinformer
 
-import "flag"
+import (
+	"flag"
+
+	corev1 "k8s.io/api/core/v1"
+)
 
 type Config struct {
-	KubeletIPAddr              string
-	KubeletHTTPPort            int
-	KubeletSyncIntervalSeconds int
-	KubeletSyncTimeoutSeconds  int
+	KubeletPreferredAddressType string
+	KubeletSyncIntervalSeconds  int
+	KubeletSyncTimeoutSeconds   int
+	ApiServerSyncTimeoutSeconds int
 }
 
 func NewDefaultConfig() *Config {
 	return &Config{
-		KubeletIPAddr:              "localhost",
-		KubeletHTTPPort:            10255,
-		KubeletSyncIntervalSeconds: 1,
-		KubeletSyncTimeoutSeconds:  3,
+		KubeletPreferredAddressType: string(corev1.NodeInternalIP),
+		KubeletSyncIntervalSeconds:  30,
+		KubeletSyncTimeoutSeconds:   3,
+		ApiServerSyncTimeoutSeconds: 10,
 	}
 }
 
 func (c *Config) InitFlags(fs *flag.FlagSet) {
-	fs.StringVar(&c.KubeletIPAddr, "KubeletIPAddr", c.KubeletIPAddr, "Kubelet IP Address.")
-	fs.IntVar(&c.KubeletHTTPPort, "KubeletHTTPPort", c.KubeletHTTPPort, "Kubelet HTTP httpPort.")
+	fs.StringVar(&c.KubeletPreferredAddressType, "KubeletPreferredAddressType", c.KubeletPreferredAddressType, "Kubelet endpoint preferred address type, options are Hostname|InternalIP|ExternalIP|InternalDNS|ExternalDNS")
```

Comment:
Usage: The node address types to use when determining which address to use to connect to a particular node.